File: radar-commons/src/main/java/org/radarbase/util/TimedInt.java
package org.radarbase.util;

public class TimedInt implements TimedVariable {
    public final int value;
    private final long expiry;

    public TimedInt(int value, long validity) {
        expiry = System.currentTimeMillis() + validity * 1000L;
        this.value = value;
    }

    @Override
    public boolean isExpired() {
        return expiry < System.currentTimeMillis();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        TimedInt other = (TimedInt)o;
        return value == other.value
                && expiry == other.expiry;
    }

    @Override
    public int hashCode() {
        return value;
    }
}

File: radar-commons/src/main/java/org/radarbase/util/TimedInt.kt
package org.radarbase.util

class TimedInt(
    val value: Int,
    cacheConfig: TimeoutConfig,
) : TimedVariable(cacheConfig) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        other as TimedInt
        return value == other.value && expiry == other.expiry
    }

    override fun hashCode(): Int = value
}
